pkg/cmd/util/clientcmd/factory.go

```diff
@@ -192,9 +192,17 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
 			if bopts.Version != nil {
-				// should --version work with builds at all?
 				return nil, errors.New("cannot specify a version and a build")
 			}
+			if bopts.Previous {
```

Comment:
i could see an argument either way here.  I think there's an argument for simple server side apis (get logs for a given build) that clients can build complex flows (determine which build to get logs for based on arguments) around.

but it is also true that if we think multiple clients will want to use this function they'd each have to reimplement it.